### PR TITLE
Add privacy notice

### LINKED ISSUE #61: Add privacy information
As we want to be able to present NGSP to a larger audience, we should add some information about how we process data. Points that might be interesting to users of our app
* We store all information they enter (name, vote, …) on a server for up to 18 hours in order to be able to continue sessions
* We do not store other user-identifiable information
* Sessions are not private–everyone who knows the link can join. But it is not possible to join without being noticed.
From a legal perspective, more information might be necessary. From discussions it should probably look something like this:

privacy

The information should be presented in a way that looks nice within the app and provide an easy way of getting back.

### PR DESCRIPTION
**Pull request by lukastaegert**
This is a first draft for adding a privacy notice. This still needs legal review and may me incomplete. Also, wording improvements are welcome.

Bildschirmfoto 2021-09-10 um 15 05 40
Bildschirmfoto 2021-09-10 um 15 05 49

Resolves #61

### CODE REVIEW

**Review by ThiloAschebrock — COMMENTED** [review]
This definitively fulfills its purpose. However, the current design in a long column does not feel optimal to me:

- On a normal screen, there is plenty of horizontal screen to fit all of the three columns next to each other.
- Clicking on the link to view the privacy details, leads to a big layout jump of the all already visible elements to the top.

Suggestion:
- Use a grid-layout that auto fills the available space so that the four boxes are displayed in a row and only break into multiple rows if the screen width is decreased.
- Instead of centering the normal login screen, place it at fixed fraction of the view port height. This way it will not jump up if content below appears.
-> yes it jumps, but would left it to an improvement, i dont expect to many people reading it regulary

File: frontend/src/Components/LoginPage/LoginPage.tsx [hunk 1]
```diff
@@ -75,6 +76,9 @@ export const LoginPage = connectToWebSocket(({ socket }) => {
       <a href={TNG_URL} target="_blank" class={classes.logo}>
         <img src={tngLogo} alt={ALT_TNG_LOGO} class={classes.logoImage} />
       </a>
+      <div class={classes.privacyNotice}>
+        <PrivacyNoticeContainer />
```
Comment (ThiloAschebrock) [inline frontend/src/Components/LoginPage/LoginPage.tsx:80]: Can we remove the div-wrapper and pass the class to `PrivacyNoticeContainer` direclty?

File: frontend/src/Components/PrivacyNoticeContainer/PrivacyNoticeContainer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { render } from '@testing-library/preact';
+import { PrivacyNoticeContainer } from './PrivacyNoticeContainer';
+import { fireEvent } from '@testing-library/preact';
+
+describe('The PrivacyNoticeContainer', () => {
+  it('displays the privacy notice when clicked', () => {
+    const { getByText, queryAllByText } = render(<PrivacyNoticeContainer />);
+    expect(queryAllByText('Processing of Personal Data')).toHaveLength(0);
```
Comment (ThiloAschebrock) [inline frontend/src/Components/PrivacyNoticeContainer/PrivacyNoticeContainer.test.tsx:8]: I am only familiar with `@testing-library/react`. With that library, I would have written:

```typescript
expect(queryByText('Processing of Personal Data')).not.toBeInTheDocument();
```
This doesn't work. Can you explain while the interface is a bit different for `preact`?

File: frontend/src/Components/PrivacyNoticeContainer/PrivacyNoticeContainer.test.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { render } from '@testing-library/preact';
+import { PrivacyNoticeContainer } from './PrivacyNoticeContainer';
+import { fireEvent } from '@testing-library/preact';
```
Comment (ThiloAschebrock) [inline frontend/src/Components/PrivacyNoticeContainer/PrivacyNoticeContainer.test.tsx:3]: '@testing-library/preact' is duplicated.

**Review by andreashille — CHANGES_REQUESTED** [review]
We should clarify the Note with legal, i‘m on it.

**Review by ThiloAschebrock — APPROVED** [review]

**Review by andreashille — APPROVED** [review]

**Comment by lukastaegert** [comment]
Sorry, kind of missed the review comments on this one. One major issue I have is that while the entire page is in English, the privacy notice is now in German, which feels both inconsistent and unprofessional to me. Do we have an English translation or should we just translate it ourselves?

**Pull request merged**